csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs

```diff
@@ -91,48 +97,37 @@ public DependencyManager(string srcDir, IDependencyOptions options, ILogger logg
                     progressMonitor.MissingNuGet();
                 }
 
-                var restoredProjects = RestoreSolutions(solutions);
+                var restoredProjects = RestoreSolutions(solutions, out var assets1);
                 var projects = allProjects.Except(restoredProjects);
-                RestoreProjects(projects);
-                DownloadMissingPackages(allNonBinaryFiles);
-            }
+                RestoreProjects(projects, out var assets2);
 
-            var existsNetCoreRefNugetPackage = false;
-            var existsNetFrameworkRefNugetPackage = false;
-            var existsNetstandardLibRefNugetPackage = false;
-            var existsNetstandardLibNugetPackage = false;
+                var dependencies = Assets.GetCompilationDependencies(progressMonitor, assets1.Union(assets2));
+
+                var paths = dependencies
+                    .Select(d => Path.Combine(packageDirectory.DirInfo.FullName, d))
+                    .ToList();
+
+                dllPathNames.AddRange(paths);
```

Comment:
I think we should log all the subfolder names in `packageDirectory` that are not added to `dllPathNames`. This might help us in identifying missing dependencies.

Reply:
Yes, that is a good idea. I have improved the solution a bit to collect dependency information in a new object (Dependencies). This will also be useful, when we improve the solution to also cater for frameworks.
It turns out that in asset files forward slashes `/` are used in package names and paths as directory separators (no matter the which OS this runs on) - the solution now explicitly takes this into account.